Source/Mac/Tools/build-assocphrase.py

```diff
@@ -3,66 +3,66 @@
 
 import sys
 
-def comp(a,b):
-	if b[1] < a[1]:
-		return -1
-	if b[1] > a[1]:
-		return 1
-	return 0
-
-if __name__=='__main__':
+
+if __name__ == "__main__":
     if len(sys.argv) != 3:
-    	sys.exit('usage: assocphrase.py input output')
-    	
+        sys.exit("usage: assocphrase.py input output")
+
     try:
-        handle = open(sys.argv[1], "r")
-        output = open(sys.argv[2], "w")
+        handle = open(sys.argv[1], "rb")
+        output = open(sys.argv[2], "wb")
     except IOError as e:
         print(e)
         sys.exit(1)
 
     phrases = {}
 
-    output.write('''%gen_inp
+    output.write(
+        """%gen_inp
 %cname 聯想詞
 %ename Associated Phrases
 %keyname begin
 %keyname end
 %chardef begin
-''')
+""".encode(
+            "utf-8"
+        )
+    )
 
     while True:
         line = handle.readline()
         if not line:
-    		break
-        if line[0] == '#':
-        	continue
+            break
+        if line[0] == "#":
+            continue
 
         elements = line.rstrip().split()
-        k = elements[0].decode('utf-8')
+        k = elements[0].decode("utf-8")
 
         if len(k) == 1:
-        	continue
+            continue
 
         v = int(elements[1])
         head = k[0]
         tail = k[1:]
 
         if head in phrases:
-        	phrases[head].append((tail, v))
+            phrases[head].append((tail, v))
         else:
-        	phrases[head] = [(tail, v)]
+            phrases[head] = [(tail, v)]
 
     handle.close()
 
+    for k in sorted(phrases.keys(), key=lambda x: x.encode("utf-8")):
```

Comment:
It seems the only difference is now we sort the keys, right?

Reply:
Yes, and we sort it by the raw bytes on purpose, so that the sorting is not influenced by any locale or collation environment settings.